ecmd-core/capi/ecmdDataBufferBase.C

```diff
@@ -1109,6 +1117,62 @@ uint32_t ecmdDataBufferBase::reverse() {
   return rc;
 }
 
+
+// This is really the same code that reverse() uses above, except it's calling fast_reverse_bytes() instead of fast_reverse32()
+// It also has an added check for byte alignment.
+uint32_t ecmdDataBufferBase::byteReverse() {
+  uint32_t rc = ECMD_DBUF_SUCCESS;
+  uint32_t l_words=0;
+  uint32_t l_slop =  (iv_NumBits % UNIT_SZ);
+
+  if (iv_NumBits % 8)
+  {
+      ETRAC("Buffer must be byte aligned to use byteReverse()");
+      RETURN_ERROR(ECMD_DBUF_INVALID_ARGS);
```

Comment:
I think ecmd is inconsistent in how it requires byte alignment.  Some methods continue on with the modification and presenting the user with a message, others do not.  Is the behavior here to actually fail?  Or should it attempt on a byte aligned buffer excluding the extras?

Reply:
My intent is to actually fail if the buffer is not byte aligned, as it might lead to other issues for the calling code.
I added the expectation to the doxygen as well. 

```
 /**
   * @brief Byte reverse entire buffer - buffer must be byte-aligned
   * @retval ECMD_DBUF_SUCCESS on success
   * @retval ECMD_DBUF_INVALID_ARGS if buffer is not byte-aligned
   */
```